scripts/inject-relup.escript

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env escript
+
+%% This script injects implicit relup instructions for emqx applications.
+
+-mode(compile).
+
+-define(ERROR(FORMAT, ARGS), io:format(standard_error, "[inject-relup] " ++ FORMAT ++ "~n", ARGS)).
+-define(INFO(FORMAT, ARGS), io:format(user, "[inject-relup] " ++ FORMAT ++ "~n", ARGS)).
+
+usage() ->
+    "Usage: " ++ escript:script_name() ++ " <path-to-release-dir> <release-vsn>".
+
+main([RelRootDir, CurrRelVsn]) ->
+    case filelib:is_dir(filename:join([RelRootDir, "releases"])) andalso
+         filelib:is_dir(filename:join([RelRootDir, "lib"])) of
+        true ->
+            EmqxAppVsns = get_emqx_app_vsns(RelRootDir),
```

Comment:
this is perhaps also no need for this versions collection,
as we aim to provide this feature for emqx app version 4.4.2
we can check if the upgrade to version is `>= 4.4.2` then there is a `emqx_relup` module, otherwise not.

Reply:
Because the only thing I know is the ReleaseVsns when I am processing the relup file, I need the mapping from ReleaseVsns to EmqxAppVsns, e.g.:

```
#{ 4.4.0 => 4.4.1, %% the emqx vsn of release 4.4.0 is 4.4.1.
     4.4.1 => 4.4.2 %% the emqx vsn of release 4.4.1 is 4.4.2.
}
```
With the EmqxVsn in place, I can then construct the `load_object_code` instruction as:

```
{load_object_code, {emqx, EmqxVsn, [emqx_relup]}}
```

Reply:
If I understand it correctly, all we need to know is which app versions are ‘before’ emqc_relup module was added.
— it’s pretty static info which will not change in the future.

Reply:
So should we maintain a RelVsn -- EmqxVsn mapping file in the repo? So we don't need to hard code it in this script, nor we need to change this this script every time we release a new version.

Reply:
I imagine this will be only applicable for 4.4.

If there’s a 4.5, emqx_relup is already there since day one.